src/main/java/picard/analysis/GcBiasMetricsCollector.java

```diff
@@ -141,7 +141,7 @@ public void acceptRecord(final GcBiasCollectorArgs args) {
                 }
                 return;
             }
-            if (!rec.getReadUnmappedFlag()) {
+            if (!rec.getReadUnmappedFlag() && args.getRef() != null) {
```

Comment:
hmmm. So the record is mapped, and a reference is provided, but somehow the record index is -1 ? how does that happen? I'm concerned that this is making the code too lenient. Could you explain why this should not, at the very least, depend on validation stringency?

Reply:
I can't explain how this happens, but it's an observation from a bam file "in the wild" 
I don't think the code should ever deref a null value. I could see an argument for failing with a validation error if stringency is set to strict.  I don't see much danger in skipping such strange records in this situation.

Reply:
I guess my point is that this "wild" bam might be invalid. in which-case I'd like to know that ValidateSamFile fails on it. There are infinite ways in which file can be invalid, and putting in protections for all the strange ways will take too much effort to maintain...

Reply:
There are a bunch of validation messages in the #1237 . Are these the validations you're referring to.
I guess this is a judgement call - worth the extra complexity to avoid the null deref or not?  Let me know if you need more from me.

Reply:
OK. I see. 

so the validation errors will stop the program from running in strict mode, but you'd like it to produce **something** in the less strict validation modes...can you provide a test for this?

Reply:
@bwlang Please add a small test, and log the fact that you are skipping this read (as a warning, but only a few times and then stop)